docs/README.md

```diff
@@ -641,6 +643,46 @@ With proper concurrency logic in place, the latest run might have been cancelled
           workflow: my-workflow.yml
 ```
 
+### github-deployment-create and github-deployment-status-update
+
+These actions create a [GitHub deployment](https://docs.github.com/en/rest/deployments/deployments) and allow updating its status. That can be useful to track progression on a workflow pipeline.
+
+Sample usage:
+
+```yaml
+permissions:
+  deployments: write # This is required for deployment statuses management
+
+jobs:
+  job:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Create Deployment
+        id: create-deployment
+        uses: Alfresco/alfresco-build-tools/.github/actions/github-deployment-create@ref
+        with:
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          environment: my_gh_environment
+          state: in_progress
+
+      - name: Other Steps
+
+      - name: Update Deployment State to failure
+        if: always() && failure() && steps.create-deployment.outcome == 'success'
+        uses: Alfresco/alfresco-build-tools/.github/actions/github-deployment-status-update@ref
+        with:
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          deployment-id: ${{ steps.create-deployment.outputs.id }}
+          state: failure
+
+      - name: Update Deployment State to success
+        uses: Alfresco/alfresco-build-tools/.github/actions/github-deployment-status-update@ref
```

Comment:
```suggestion
        if: steps.create-deployment.outcome == 'success'
        uses: Alfresco/alfresco-build-tools/.github/actions/github-deployment-status-update@ref
```

otherwise it could cause an additional failure?

Reply:
it is implied: if the step was not successful then a step without any condition will be skipped